### DISCUSSION TITLE
Filter records [May 12]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5677] [Discord Threads]
> 🙋‍♂️ **Rahim Dobani**

**Comment by o1lab** [comment]
> 🙋‍♂️ **Dstala | NocoDB Team**  
#### Assuming your question is about use of API, you can try out swagger spec to quickly experiment.   
####    
#### For a sample table with SingleLineText "Title" column on "Sheet-1" table, you can use query    
#### http://localhost:8080/api/v1/db/data/v1/mon_mothma/Sheet-1?where=%28Title%2Ceq%2Ca%29&limit=25&shuffle=0&offset=0   
####    
#### condition specified is where=Title, eq,a  
#  
> 🙋‍♂️ **Dstala | NocoDB Team**  
####   
![image0](https://cdn.discordapp.com/attachments/1093846449617375302/1094809230130815076/Screenshot_2023-04-10_at_7.47.27_AM.png)  
![image1](https://cdn.discordapp.com/attachments/1093846449617375302/1094809230353117224/Screenshot_2023-04-10_at_7.50.46_AM.png)  
#  
> 🙋‍♂️ **sukhan**  
#### with DateTime fields should this work? This construction - (DateTimeAction,eq,today) gives null result, although there are records with today's date.  
#  
> 🙋‍♂️ **Dstala | NocoDB Team**  
#### it expects explicit date i believe  
#  
> 🙋‍♂️ **sukhan**  
#### for the exact date there is another expression, but it doesn't work either. the question is it something I'm doing wrong, or now filtering on fields like date_time doesn't work in principle? I'm testing in swagger, examples from help. what could be wrong?  
![image0](https://cdn.discordapp.com/attachments/1093846449617375302/1095240841079042048/filter_date_exact.png)  
#  
> 🙋‍♂️ **Dstala | NocoDB Team**  
#### i tried them out & it looks buggy. will discuss with the team & update on this  
#  
> 🙋‍♂️ **Dstala | NocoDB Team**  
#### my bad. I had tried without braces included. following worked for me   
#### condition : `(title5,eq,exactDate,2023-04-12)`   
#### http://localhost:8080/api/v1/db/data/v1/shallow_mule/Sheet-1?where=%28Title%2Ceq%2Ca%29&limit=25&shuffle=0&offset=0   
####    
#### condition : `(title5,eq,today)`    
#### http://localhost:8080/api/v1/db/data/v1/shallow_mule/Sheet-1?where=%28title5%2Ceq%2Ctoday%29&limit=25&shuffle=0&offset=0  
#  
> 🙋‍♂️ **sukhan**  
#### There is another question - if the field by which the filter is set is only Date (e.g. 2023-04-10), then constructs like (colName,eq,yesterday) and (colName,eq,exactDate,2022-02-02) work. But if the field is of type Date and Time (e.g. 2021-07-20T07:57:00.000Z) - then the filter does not work.  
#  
> 🙋‍♂️ **Dstala | NocoDB Team**  
#### for date time, it's expected to include time; so condition for exact date will have to be built using "and" operator   
####    
#### `(title5,gt,exactDate,2023-04-12 00:00)~and(title5,lte,exactDate,2023-04-12 23:59)`   
####    
#### http://localhost:8080/api/v1/db/data/v1/upper_cod/Sheet-1?where=%28title5%2Cgt%2CexactDate%2C2023-04-12%2000%3A00%29~and%28title5%2Clte%2CexactDate%2C2023-04-12%2023%3A59%29&limit=25&shuffle=0&offset=0  
#  
> 🙋‍♂️ **Dstala | NocoDB Team**  
#### gt - greater than;    
#### lte - less than or equal to;  
#  
> 🙋‍♂️ **sukhan**  
#### thank you very much. honestly - not obvious. it would be good to add it to the reference guide.  
#  
> 🙋‍♂️ **Dstala | NocoDB Team**  
#### <@751772400978231299> agree. we are working on improving our api docs to include more examples  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1093846449617375302)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
